### PR TITLE
fix routes for user's course

### PR DESCRIPTION
**Pull request by kian1213**
Hi sir, yesterday I had trouble for doing the adding of supervisor for course, I get the root cause of problems, first I want to fix the routes for every User course created. Please review my pull request, Thanks po.

### CODE REVIEW

**Comment by kian1213** [comment]
I'm coding at home, can't ask Grace for comment ok.

File: app/models/course.rb [hunk 1]
```diff
@@ -11,6 +11,26 @@ class Course < ActiveRecord::Base
   validates :end_date, presence: true
   validates :status, presence: true
 
-  accepts_nested_attributes_for :course_subjects, reject_if:
-    lambda {|a| a[:subject_id].blank?}, allow_destroy: true
+  accepts_nested_attributes_for :course_subjects, allow_destroy: true
+
+  def get_subjects
```
Comment (dieunb) [inline app/models/course.rb]: no need it.

File: app/views/courses/edit.html.erb [hunk 2]
```diff
@@ -13,7 +13,70 @@
...
+          <table class="table table-striped table-advance course-table">
+            <thead>
+              <tr>
+                <th><%= t ".select" %></th>
+                <th><%= t ".subject_title" %></th>
+                <th><%= t ".instruction "%></th>
+              </tr>
+            </thead>
+            <tbody>
+              <%= f.fields_for :course_subjects, @course.get_subjects do |builder| %>
```
Comment (dieunb) [inline app/views/courses/edit.html.erb]: you can build course_subjects in controller

Comment (kian1213) [inline app/views/courses/edit.html.erb]: How sir? please give me the idea in this one.. I need all the subjects but also I need to build. So I thought that adding the function in model to get existing subjects to add and edit is okay.

File: app/views/courses/edit.html.erb [hunk 3]
```diff
@@ -13,7 +13,70 @@
...
+          <table class="table table-striped table-advance course-table">
+            <thead>
+              <tr>
+                <th><%= t ".select" %></th>
+                <th><%= t ".subject_title" %></th>
+                <th><%= t ".instruction "%></th>
+              </tr>
+            </thead>
+            <tbody>
+              <%= f.fields_for :course_subjects, @course_subjects do |builder| %>
```
Comment (dieunb) [inline app/views/courses/edit.html.erb]: `<%= f.fields_for :course_subjects do |builder| %>`

File: app/controllers/courses_controller.rb [hunk 4]
```diff
@@ -33,7 +33,24 @@ def course_params
       :status, course_subjects_attributes: [:id, :course_id, :subject_id, :_destroy]
   end
 
-  def load_all_subjects
+  def load_all_references
```
Comment (dieunb) [inline app/controllers/courses_controller.rb]: it is too complex

**Comment by kian1213** [comment]
The namespace admin has this CRUD too, if this successfully merged, Ill then start transferring this to the namespace,, I just might update too much files if I include in this pull. thanks

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 class CoursesController < ApplicationController
   load_and_authorize_resource
-  before_action :load_all_subjects, on: [:new, :edit]
+  before_action :load_subjects, only: [:new]
+  before_action :load_subjects, only: [:new, :edit]
```
Comment (dieunb) [inline app/controllers/courses_controller.rb]: duplicated above

**Pull request merged**